decisions/0001-bundle-granularity.rst

```diff
@@ -46,56 +46,145 @@ XBlock Directories are a client convention and not a core Blockstore primitive.
+        <!-- A second usage of the same problem bank -->
+        <xblock-include definition="prob_bank5/unit/3problems" usage="second_set" />
+    </unit>
+
+In this example, if the included ``3problems`` unit contains child includes
+like::
+
+    <xblock-include definition="problem/problem1" usage="p1" />
+    ...
+
+Then in the original example, the usage ID ``p1`` would not be unique, because
+its parent unit is included twice. So the learning context will have to prefix
+the usage hint ``p1`` in this case, producing unique usage keys like
+``first_set-p1`` for the first usage and ``second_set-p1`` for the second.
+
+---------------------------------
+Proposal: Content Library Bundles
+---------------------------------
+
+A content library bundle is an OLX bundle with additional constraints, that is
+used to represent the OLX data of a content library.
+
+As with OLX Bundles in general, the conventions defined here are not implemented
+nor enforced by Blockstore in any way, however they are documented here for the
+sake of having a central reference.
+
+A content library bundle is an OLX bundle that holds a collection of one or more
+XBlocks. Each XBlock definition (OLX ``definition.xml`` file) in the content
+library is either a "top-level block" or is a child of exactly one other XBlock
+definition in the same bundle. A content library bundle does not allow a single
+definition to have multiple different usages (multiple <xblock-include />
+elements referencing the same definition). However, an XBlock definitions from
+any other linked OLX bundle, such as another content library, may be used
+multiple times::
+
+    <unit display_name="">
+        <!-- Normal usage of a definition in the same library bundle -->
+        <xblock-include definition="problem/problem1" />
+
+        <!-- The following is not allowed in a content library because each
+        definition in a content library may only be used once:
+        <xblock-include definition="problem/problem1" usage="other-usage" />
+        -->
+
+        <!-- However, XBlock definitions from linked bundles may be used freely -->
+        <xblock-include definition="linked_problem_bank/problem/problemB" usage="probB" />
+        <xblock-include definition="linked_problem_bank/problem/problemB" usage="probB-alt" />
+    </unit>
+
+In a content library bundle, the ``usage="..."`` attribute must not be specified
+when including an XBlock definition from the same bundle, but must be specified
+when including an XBlock definition from a linked bundle. This makes the
+implementation of content libraries considerably simpler.
+
+------------------------
+Proposal: Course Bundles
+------------------------
+
+The format of course bundles has not yet been finalized but will likely be
+an OLX bundle that includes some sort of additional "outline" file, such as
+``course-outline.json`` which specifies how the various XBlock definitions it
```

Comment:
FWIW, I think that we're going to need multiple files if we're going to support sub-section and section level publishing, but I'm happy to punt that discussion to later and leave this paragraph as-is.